src/base/ApplicationDynamoDBTable.ts

```diff
@@ -38,6 +43,11 @@ export interface ApplicationDynamoDBProps {
   tableConfig: ApplicationDynamoDBTableConfig;
   readCapacity?: ApplicationDynamoDBTableAutoScaleProps;
   writeCapacity?: ApplicationDynamoDBTableAutoScaleProps;
+  // If capacityMode is ON_DEMAND, the DynamoDB table will have on-demand capacity. By default this is PROVISIONED.
+  // On-demand capacity mode is capable of serving thousands of requests per second without capacity planning.
+  // https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/HowItWorks.ReadWriteCapacityMode.html
+  // The readCapacity and writeCapacity properties should not be used if capacityMode is set to ON_DEMAND.
```

Comment:
@mmiermans can you throw an error if a consumer sets these values while having ON_DEMAND set? That will cause the terraform to not synthesize and prevent some errors :)

Reply:
Sri and I think Terraform should do the validation here?

Reply:
@mmiermans It may, but i wouldn't count on it. 

There are also benefits of doing it in typescript here. If you do it here, when a consumer tries to compile into terraform it will error, which is much quicker then waiting for a terraform apply to error out.

Also we can make nicer error messages :)